python/ray/data/datasource/file_based_datasource.py

```diff
@@ -497,11 +502,24 @@ def _resolve_paths_and_filesystem(
                     _encode_url(path), filesystem
                 )
                 resolved_path = _decode_url(resolved_path)
+            elif "Unrecognized filesystem type in URI" in str(e):
+                scheme = urllib.parse.urlparse(path, allow_fragments=False).scheme
+                if scheme in ["http", "https"]:
```

Comment:
whats the current user experience after this patch if fsspec is not installed?

Reply:
@richardliaw - yeah I think we'd better to give user a clearer message to install fsspec, this is new error message after this PR now:

```
ray::_get_read_tasks() (pid=72537, ip=127.0.0.1)
  File "/Users/chengsu/ray/python/ray/data/read_api.py", line 1123, in _get_read_tasks
    reader = ds.create_reader(**kwargs)
  File "/Users/chengsu/ray/python/ray/data/datasource/file_based_datasource.py", line 206, in create_reader
    return _FileBasedDatasourceReader(self, **kwargs)
  File "/Users/chengsu/ray/python/ray/data/datasource/file_based_datasource.py", line 343, in __init__
    paths, self._filesystem = _resolve_paths_and_filesystem(paths, filesystem)
  File "/Users/chengsu/ray/python/ray/data/datasource/file_based_datasource.py", line 539, in _resolve_paths_and_filesystem
    raise ImportError(f"Please install fsspec to read files from HTTP.")
ImportError: Please install fsspec to read files from HTTP
```

Reply:
great!